modules/product/src/main/java/com/opengamma/strata/product/SecurityId.java

```diff
@@ -0,0 +1,205 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ * 
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product;
+
+import java.io.Serializable;
+
+import org.joda.beans.PropertyDefinition;
+import org.joda.convert.FromString;
+import org.joda.convert.ToString;
+
+import com.opengamma.strata.basics.market.ReferenceData;
+import com.opengamma.strata.basics.market.ReferenceDataId;
+import com.opengamma.strata.basics.market.ReferenceDataNotFoundException;
+import com.opengamma.strata.basics.market.Resolvable;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.id.StandardId;
+
+/**
+ * An identifier for a security.
+ * <p>
+ * This identifier is used to obtain a {@link ReferenceSecurity} from {@link ReferenceData}.
+ * <p>
+ * A security identifier uniquely identifies a security within the system.
+ * A real-world security will typically have multiple identifiers.
+ * The only restriction placed on the identifier is that it is sufficiently
+ * unique for the reference data lookup. As such, it is acceptable to use
+ * an identifier from a well-known global or vendor symbology.
+ */
+public final class SecurityId
+    implements ReferenceDataId<ReferenceSecurity>, Resolvable<ReferenceSecurity>, Serializable {
+
+  /** Serialization version. */
+  private static final long serialVersionUID = 1L;
+
+  /**
+   * The identifier, expressed as a standard two-part identifier.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final StandardId standardId;
+
+  //-------------------------------------------------------------------------
+  /**
+   * Obtains an instance from a scheme and value.
+   * <p>
+   * The scheme must be non-empty and match the regular expression '{@code [A-Za-z0-9:/+.=_-]*}'.
```

Comment:
Could the restriction on the scheme be stated without requiring the user to parse a regular expression in their head?